types/stoppable/index.d.ts

```diff
@@ -2,17 +2,20 @@
 // Project: https://github.com/hunterloftis/stoppable
 // Definitions by: Eric Byers <https://github.com/EricByers>
 //                 John Plusjé <https://github.com/jplusje>
+//                 Ilia Baryshnikov <https://github.com/qwelias>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
 
 /// <reference types="node" />
-import { Server } from 'http';
+import { Server as NetServer } from 'net';
+import { Server as HttpServer } from 'http';
+
+declare function stoppable <S extends NetServer>(server: S, grace?: number): stoppable.StoppableServer<S>;
 
 declare namespace stoppable {
-  interface StoppableServer extends Server {
-    stop(callback?: (e: Error, gracefully: boolean) => void): void;
-  }
+    type StoppableServer<S = HttpServer> = S & {
```

Comment:
Does this need to be a type alias rather than an interface? Interfaces generally give better error messages.

Reply:
@jessetrinity How can I make it a generic interface that extends `S` and also keep it backward compatible?